public/app/features/dashboard/components/PanelEditor/PanelEditor.tsx

```diff
@@ -117,15 +118,20 @@ export class PanelEditorUnconnected extends PureComponent<Props> {
     this.forceUpdate();
   };
 
-  onDiscard = () => {
-    this.props.setDiscardChanges(true);
-
+  onBack = () => {
     locationService.partial({
       editPanel: null,
       tab: null,
     });
   };
 
+  onDiscard = () => {
+    this.props.setDiscardChanges(true);
+    this.props.panel.hasChanged = false;
+
+    this.onBack();
+  };
+
```

Comment:
This code seems very similar to the code in the `exitPanelEditor` Thunk, is there a way we can move this into a separate thunk that can be called from here and from the `exitPanelEditor`.

Reply:
Good idea!